Added concept and howto for Flink application (#1706)
aiven/devportal: docs/products/flink/concepts/flink-applications.rst

Features of Aiven for Apache Flink applications
-----------------------------------------------

Some of key feature of Aiven for Flink applications include: 

* **Intuitive Interface**: The `Aiven Console <https://console.aiven.io/>`_ provides a user-friendly, guided wizard to help you quickly build and deploy your applications.
* **Source and sink tables** definition with **SQL autocomplete**: You can create source and sink tables, the SQL autocomplete feature allows you to start from a specific set of pre-filled parameters depending on the type of connector.
* **Interactive query**: The interactive query feature allows you to validate and preview your data in the table or SQL tranformation before deploying it.
* **Start and stop**: you can start and stop applications stop and start anytime.
* **Automatic Versioning**: on every edit of the table or data transformation definition, a new application versions is automatically generated allowing you to keep track of the changes and revert to a previous version if necessary.
* **Savepoints**: You can stop your application with a savepoint, allowing you to restart it from the previous state at a later time.

> * **Scalability:** You can scale your cluster depending on the demands of your specific application by parallelizing the applications into multiple tasks and executing them concurrently in a cluster.

Reviewer: I believe we need to discuss this in more detail:
* Scalability of the cluster is about Apache Flink itself (not at application level)
* Change the level of parallelism is at application level, but needs a separate discussion

Author: Got it, thanks. Updated the text.

Limitations 
------------

> * **Concurrent applications:** You can create and run up to 4 concurrent applications in the Aiven for Apache Flink service.

Reviewer: Should this be `10`?

Author: That is right. Updated this.

> * **Savepoints:** You can store up to 10 savepoints history per application deployment, allowing you to easily roll back to previous versions of your application if necessary. If you exceed this limit, you must clear the savepoints history before creating a new application version.

Reviewer: ```suggestion
* **Savepoints:** You can store up to 10 savepoints history per application deployment, allowing you to start from a previous state if necessary. If you exceed this limit, you must clear the savepoints history before creating a new application deployment.
```

Author: Fixed
